src/player_mechanics.rs

```diff
@@ -98,3 +101,156 @@ pub fn end_attack(mut commands: Commands, query: Query<(&BasicAttack, Entity, &D
+        let dark_attack = asset_server.load("dark_attack.png");
+        let nature_attack = asset_server.load("nature_attack.png");
+        let air_attack = asset_server.load("air_attack.png");
+        let mut attack_direction_x = 0.0;
+        let mut attack_direction_y = 0.0;
+
+        if keyboard_input.pressed(KeyCode::Right) {
+            attack_direction_x = 30.;
+        }
+
+        if keyboard_input.pressed(KeyCode::Left) {
+            attack_direction_x = -30.;
+        }
+
+        if keyboard_input.pressed(KeyCode::Up) {
+            attack_direction_y = -30.;
+        }
+
+        if keyboard_input.pressed(KeyCode::Down) {
+            attack_direction_y = 30.;
+        }
+
+        if keyboard_input.just_pressed(KeyCode::X) {
+            if attack_direction_x == 0. && attack_direction_y == 0. {
+                attack_direction_x = 10.;
+            }
+            
+            for mut delay in special_delay.iter_mut() {
+                if delay.next_action_aviable(Instant::now()) {
+                    delay.start = Instant::now();
+
+                    if attack_direction_x == 0. && attack_direction_y == 0. {
+                        attack_direction_x = 30.;
+                    }
+                    inventory.for_each(|inventory| {
+                        match inventory.p_element {
+                            Element::Darkness => {
+                                commands
+                                    .spawn_bundle(SpriteBundle {
+                                        image: dark_attack.clone(),
+                                        transform: Transform::from_xyz(position.translation.x + attack_direction_x,
+                                             position.translation.y + attack_direction_y, 0.),
+                                        ..Default::default()
+                                    })
+                                    .insert(TesselatedCollider {
+                                        image: dark_attack.clone(),
+                                        tesselator_config: TesselatedColliderConfig {
+                                            vertice_separation: 0.,
+                                            ..Default::default()
+                                        },
+                                        ..Default::default()
+                                    })
+                                    .insert(RigidBody::Sensor)
+                                    .insert(Velocity::from_linear(Vec3::default()))
+                                    .insert(BasicAttack)
+                                    .insert(Specialty{value: String::from("none")})
+                                    .insert(Delay {delay: 0.2, start: Instant::now()})
+                                    .insert(CollisionLayers::new(Layer::Projectile, Layer::Enemy))
+                                    .insert(Damage{value: 4.})
+                                    .insert(LevelEntity);
```

Comment:
You can make a bundle which includes the basic components you need, and just use .insert_bundle to add them all toghether